controllers/certificatesigningrequests_controller.go

```diff
@@ -69,21 +71,40 @@ func NewCertificateSigningRequestsReconciler(mgr manager.Manager, clusterConfig
 // Reconcile is part of the main kubernetes reconciliation loop which reads that state of the cluster for a
 // CertificateSigningRequests object and aims to move the current state of the cluster closer to the desired state.
 func (r *certificateSigningRequestsReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
-	_ = r.log.WithValues("certificatesigningrequests", req.NamespacedName)
+	log := r.log.WithValues("certificatesigningrequests", req.NamespacedName)
+
+	// Update operator condition to prevent OLM from updating WMCO while CSRs are being processed
+	if err := conditions.PatchUpgradeable(r.client, r.watchNamespace, meta.ConditionFalse,
+		conditions.UpgradeableFalseMessage, conditions.UpgradeableFalseReason); err != nil {
+		log.Info("unable to set Upgradeable condition, be cautious of automatic operator upgrades", "error", err)
```

Comment:
I suggest sending an event in addition to the log here. This applies to all places where an error is happening.

While I think not returning an error is fine, please check with the OLM team also.

Reply:
The OLM team mentioned that operators would usually retry on transient errors (e.g. connection issues). Because of the high number of places/code paths that set `Upgradeable` (multiple places in each controller), I think we mitigate that. So sending and event and logging should be fine.

Reply:
I suggest retrying given the feedback from the OLM team. Having multiple places where we set `Upgradable` is not a solution to the problem.

Reply:
After discussion with OLM and Operator SDK folks:
- will retry on error
- will wait for `Upgradeable` to be updated in `OperatorCondition` status, not just spec